app/api/2fa/check/route.ts

```diff
@@ -0,0 +1,29 @@
+import type { NextRequest } from "next/server";
+import { NextResponse } from "next/server";
+
+import prisma from "@/lib/prisma";
+
+export async function POST(req: NextRequest) {
```

Comment:
Could an attacker check if someone had 2FA enabled here? Maybe should make this an authenticated request.